### ISSUE TITLE
Duplicate elements showing when adding style to progressive-background

### ISSUE THREAD

**Issue by leah-pgdn** [MatteoGabriele/vue-progressive-image#48]
### Description

Duplicate elements showing when adding style to `progressive-background`

### Expected behavior

Show the elements inside only once

### Actual behavior

The elements get duplicated

### Environment

`npm ls vue-progressive-image`: `-- vue-progressive-image@2.5.0`

1. Operating system: Windows 10 Pro
2. Browser and version: Google Chrome - Version 66.0.3359.181 (Official Build) (64-bit)

### Reproducible Demo

I created a new vue app and placed this in the generated `Home.vue`

```
<progressive-background style="height: auto; overflow: hidden; padding: 15px" src="https://picsum.photos/200/300/?blur">
    <p style="margin: 15px 0 0 0; font-weight: bold;">JOHN DOE</p>
</progressive-background>
```

Result:
![screenshot_32](https://user-images.githubusercontent.com/20717836/40529233-8a11e504-6026-11e8-859e-059c93866e74.png)

If I remove `style` from `progressive-background` then the content will only show once

**Comment by atxy2k** [comment]
I have a similar problem, I have the next code:

```
<div class="Gallery" >
                            <div class="Gallery-item" v-for="image in files">
                                <a href="#" class="Gallery-item-link">
                                    <progressive-img :src="image.full_path" class="Gallery-item-link-img" :placeholder="image.full_path">
                                        <div slot-scope="{ visible }" v-if="visible"
                                             style="width: 100%;height: 100%;
                                                    display: flex;align-items: center;justify-content: center;background-color: #f4f5f8;">
                                            <i class="fa fa-spinner fa-pulse fa-3x fa-fw" style="font-size: 1.5em;color: #a4a2bb;"></i>
                                        </div>
                                    </progressive-img>
                                    <span class="Gallery-item-check">
                                        <i class="la la-check"></i>
                                    </span>
                                </a>
                            </div>
                        </div>
```

All images are diferent, and I add them dinamically when the gallery component are hidden, but, I get the next:

![captura 1](https://user-images.githubusercontent.com/2886154/41194878-633af6f0-6be8-11e8-90a1-c84f64c6b01a.PNG)

 The two images see similar but in debug are diferent.

![captura 2](https://user-images.githubusercontent.com/2886154/41194888-7e6071d0-6be8-11e8-97ac-337eec6a4d6e.PNG)

I tried clean my files array en reasign it, but not works, Then if I close the modal and open it again, how I do ajax request, the images see ok. 

Can you help me?

**Comment by MatteoGabriele** [comment]
I wouldn't change the style of the component itself, is probably interfering with the component internal CSS structure. 
I would use a wrapper around it if you need to do something special.

**Comment by MojoDeMedici** [comment]
@MatteoGabriele I get a weird error in my app when I wrap the progressive background around the rest of my template.

```
<progressive-background>
<section class="hero is-fullheight" >
  <div class="hero-body">
    <main-body></main-body>
  </div>
  <div class="hero-foot">
    <nav-footer></nav-footer>
  </div>
 </section>
</progressive-background>
```

**Duplicate presence of slot "default" found in the same render tree - this will likely cause render errors.**

I cannot fix this:( the slot should be named to avoid conflicts.
 The rendering process fails, and render things 2 times.

It duplicates elements exactly as above described. I do think the issue is related with the slot naming. I am going to try to mount the version in which the slot was not implemented to see if it works properly.

Get back here with the results.

@MatteoGabriele as I suspected the issue is related with the addition in the last version (2.5.0) of the slot. The version 2.4.4 works perfectly. The slot should be named because it creates conflicts with everything else when the progressive component wraps around other components. I do suggest to name it, simply because wrapping it around other things is kind of natural, especially when you are dealing with the background. Or you can use a scoped slot.

https://vuejs.org/v2/guide/components-slots.html

Note the bad behavior is clearly not present when instead the progressive component is by itself, meaning is not wrapped around other things.

This make sense?

looking forward to hearing from you back.

**Comment by leah-pgdn** [comment]
@MatteoGabriele Ok so I tried to wrap it around a div and I finally got what I wanted to happen.
```
    <div style="height: auto; overflow: hidden; padding: 15px">
        <progressive-background src="https://picsum.photos/200/300/?blur">
            <div style="padding:15px">
                <p style="margin: 15p; font-weight: bold;">JOHN DOE</p>
            </div>
        </progressive-background>
    </div>
```
I don't get why I still need to wrap `p` in a `div` inside the `progressive-background` tho. It's just too much elements for me. But it works now. Thank you so much!

**Comment by leah-pgdn** [comment]
@MatteoGabriele I also got the same error as stated by @Dinuz

**Comment by MatteoGabriele** [comment]
it's because when you apply a style to a component, this style is applied directly to the component and it will possibly override its property. In this specific case, you were modifying required property for the component to be able to function.

**Comment by MatteoGabriele** [comment]
@atxy2k @Dinuz can you both please create a separate ticket and a public repository with your issue in a simple application with vue-cli? 

thank you both for your help

**Comment by leah-pgdn** [comment]
@MatteoGabriele I see... Thanks for the explanation! :D Will be closing this ticket now.
